### PR TITLE
Version 1.3.0

### PR DESCRIPTION
**Pull request by ShooterIT**
## Version 1.3.0

License
  - Change the license from MIT to BSD-3-Clause

New features
  - Support to set separate log directory to store server logs (#155)
  - Implement PSETEX command (#156)
  - Implement MSETNX command (#165)

Improvements
  - Improve the precision of the INCRBYFLOAT/HINCRBYFLOAT command to double (#169)
  - Support +INF/-INF for the score of zset type members (#172)
  - The UNSUBSCRIBE and PUNSUBSCRIBE commands support multiple channels or patterns (#174)

Behavior changes
  - The UNSUBSCRIBE and PUNSUBSCRIBE commands have the same reply as Redis (#174)
  - The same message format as Redis for pattern subscribers when publishing message (#175)
    Message format is from 'message channel message' to 'pmessage pattern channel message'

Bug fixes
  - Fix skipping some SST files when picking compaction files (#154)
  - Handle negative expire time in set command to avoid deleting keys (#156)
  - List type command doesn't rewrite other types of data (#157)
  - Only set positive expire time when migrating key (#160)
  - Don't store an empty key for setrange command (#161)
  - Fix SETBIT/GETBIT bit offset out of range (#164)
  - Fix INCRBY/INCRBYFLOAT command operates on wrong value and precision check (#167)
  - Fix HINCRBY/HINCRBYFLOAT command operates on wrong value and precision check (#168)
  - RPOPLPUSH command doesn't operate non list type (#170)
  - LINSERT command returns -1 if don't find index (#170)
  - Fix LTRIM command range calculation when start is very large negative number (#170)
  - SMOVE command doesn't operate non set type key (#171)
  - Fix ZREMRANGEBYRANK command corrupts the size of zset (#172)
  - SUBSCRIBE and PSUBSCRIBE commands return the sum of subscription and psubscription (#174)
  - Fix PUNSUBSCRIBE command doesn't take effort (#174)
  - Allow setting slowlog-log-slower-than to -1 to disable the slowlog (#178)

### CODE REVIEW

**Review by karelrooted — DISMISSED** [review]
LGTM

File: Changelog [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+  - Implement MSETNX command (#165)
+
+Improvements
+  - Improve the precision of the INCRBYFLOAT/HINCRBYFLOAT command to double (#169)
+  - Support +INF/-INF for the score of zset type members (#172)
+  - The UNSUBSCRIBE and PUNSUBSCRIBE commands support multiple channels or patterns (#174)
+
+Behavior changes
+  - The UNSUBSCRIBE and PUNSUBSCRIBE commands have the same reply as Redis (#174)
+  - The same message format as Redis for pattern subscribers when publish message (#175)
```
Comment (git-hulk) [inline Changelog:19]: publish => publishing

File: Changelog [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+    Message foromat is from 'message channel message' to 'pmessage pattern channel message'
```
Comment (git-hulk) [inline Changelog:20]: typo: foromat=> format

File: Changelog [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+
+Bug fixes
+  - Fix skipping some SST files when pick compaction files (#154)
```
Comment (git-hulk) [inline Changelog:23]: pick => picking

File: Changelog [hunk 4]
```diff
@@ -1,3 +1,42 @@
...
+  - Handle negative expire time in set command to avoid deleting keys (#156)
+  - List type command doesn't rewrite other types data (#157)
```
Comment (git-hulk) [inline Changelog:25]: List type command doesn't rewrite other types data  => List type command doesn't rewrite other types of data?

File: Changelog [hunk 5]
```diff
@@ -1,3 +1,42 @@
...
+  - Only set positive expire time when migrate key (#160)
```
Comment (git-hulk) [inline Changelog:26]: migrate => migrating

File: Changelog [hunk 6]
```diff
@@ -1,3 +1,42 @@
...
+  - Fix INCRBY/INCRBYFLOAT command operates on wrong value and precision check (#167)
+  - Fix HINCRBY/HINCRBYFLOAT command operates on wrong value and precision check (#168)
+  - RPOPLPUSH command doesn't operate non list type (#170)
+  - LINSERT command returns -1 if don't find index (#170)
+  - Fix LTRIM command range calculation when start is very large negative number (#170)
+  - SMOVE command doesn't operate non set type key (#171)
+  - Fix ZREMRANGEBYRANK command corrupts the size of zset (#172)
+  - SUBSCRIBE and PSUBSCRIBE commands return the sum of subscription and psubscription (#174)
+  - Fix PUNSUBSCRIBE command doesn't take effort (#174)
+  - Allow to set slowlog-log-slower-than to -1 to disable the slowlog (#178)
```
Comment (git-hulk) [inline Changelog:38]: Allow to set =>  Allow setting

File: Changelog (hunk 1 above)
Comment (ShooterIT) [inline Changelog:19]:
```suggestion
  - The same message format as Redis for pattern subscribers when publishing message (#175)
```

File: Changelog (hunk 2 above)
Comment (ShooterIT) [inline Changelog:20]:
```suggestion
    Message format is from 'message channel message' to 'pmessage pattern channel message'
```

File: Changelog (hunk 3 above)
Comment (ShooterIT) [inline Changelog:23]:
```suggestion
  - Fix skipping some SST files when picking compaction files (#154)
```

File: Changelog (hunk 4 above)
Comment (ShooterIT) [inline Changelog:25]:
```suggestion
  - List type command doesn't rewrite other types of data (#157)
```

File: Changelog (hunk 5 above)
Comment (ShooterIT) [inline Changelog:26]:
```suggestion
  - Only set positive expire time when migrating key (#160)
```

File: Changelog (hunk 6 above)
Comment (ShooterIT) [inline Changelog:38]:
```suggestion
  - Allow setting slowlog-log-slower-than to -1 to disable the slowlog (#178)
```

**Review by git-hulk — APPROVED** [review]

File: Changelog [hunk 7]
```diff
@@ -1,3 +1,42 @@
+## Version 1.3.0
+-------------------------------------------------------------------------------
```
Comment (ShooterIT) [inline Changelog:2]:
```suggestion
```

**Review by git-hulk — APPROVED** [review]

**Review by karelrooted — APPROVED** [review]

**Pull request merged**